src/state/operator/selectors.js

```diff
@@ -82,14 +106,62 @@ const compareOperatorPriority = ( a, b ) => {
 		}
 	}
 
-	// Compare operators by their current chat load
 	if ( a.percentAvailable === b.percentAvailable ) {
 		if ( a.totalAvailable === b.totalAvailable ) {
 			return 0;
 		}
-		return a.totalAvailable > b.totalAvailable ? -1 : 1
+		return a.totalAvailable > b.totalAvailable ? -1 : 1;
+	}
+	return a.percentAvailable > b.percentAvailable ? -1 : 1;
+}
+
+/**
+ * Compare two operators using their status
+ *
+ * @param {Operator} a First operator to compare
+ * @param {Operator} b Second operator to compare
+ * @return {Number} -1 if operator a is available
+ *                  0 if they have the same status
+ *                  1 if operator b is available
+ */
+const compareOperatorByStatus = ( a, b ) => {
+	if ( a.status === b.status ) {
+		return 0;
 	}
-	return a.percentAvailable > b.percentAvailable ? -1 : 1
+
+	// When comparing two operators, always prioritise the one whose
+	// status is available over the one who is in reserve
+	return a.status === STATUS_AVAILABLE ? -1 : 1;
+}
+
+/**
+/* Compare function for sorting operator priority.
+ *
+ * The first operator in the resulting list should be the next one
+ * to be assigned to a chat. This is used for balancing chat load
+ * evenly across operators.
+ *
+ * @param {Operator} a First operator to compare
+ * @param {Operator} b Second operator to compare
+ * @return {Number} -1 if operator a should come first
+ *                  0 if both operators are equal priority
+ *                  1 if operator b should come first
+ */
+const compareOperatorPriority = ( a, b ) => {
+	// Prioritize operators who are requesting a chat
+	const requestForChatComparison = compareByRequestForChat( a, b );
+	if ( requestForChatComparison !== 0 ) {
+		return requestForChatComparison;
+	}
+
+	// When comparing two operators, always prioritise the one whose
+	// status is available over the one who is in reserve
+	const statusComparison = compareOperatorByStatus( a, b );
+	if ( statusComparison !== 0 ) {
+		return statusComparison;
+	}
+
+	return compareOperatorsByChatLoad( a, b );
```

Comment:
Another way we could write this function:

```js
const compareOperatorPriority = ( a, b ) => {
	// A function to convert a comparator to the calculated priority for the operators
	const mapComparators = map( f => f( a, b ) );
	// A function to find the first priority where both operators are not equal
	const firstPriority = find( p => p !== 0 );

	// List of priority comparators. The order of this list is important -
	// the first non-zero comparator will determine the priority for the
	// operators.
	const priorities = mapComparators( [
		compareByRequestForChat,
		compareOperatorByStatus,
		compareOperatorsByChatLoad,
	] );
	return firstPriority( priorities ) || 0;
}
```

What this will do is go through the list of comparators, and return the result of the first non-zero priority. So if `compareByRequestForChat` returns zero, it'll fall back to `compareOperatorByStatus` and so on.

This will let us clearly see and add/drop/reorder priority calculations in the future by simply changing the array elements.

This makes use of ramda's `find` and `map`, but could be done in lodash too (although I think ramda makes it a bit easier to read by making it possible to define a specialist `find`er and `map`per).